src/nodes/branch.rs

```diff
@@ -193,6 +201,50 @@ impl<'a> BranchNodeRef<'a> {
     }
 }
 
+/// Iterator over branch node children.
+#[derive(Debug)]
+pub struct BranchChildrenIter<'a> {
+    pos: u8,
+    end: u8,
+    stack_ptr: usize,
+    state_mask: &'a TrieMask,
+    stack: &'a [Vec<u8>],
```

Comment:
looks like:
- `pos` and `end` can be just `range: Range<u8> = CHILD_INDEX_RANGE` which is also an iterator
- `stack` and `stack_ptr` can be `slice::Iter` from `node.stack[node.first_child_index()..].iter()`

Reply:
collapsed pos and end, stack ptr can't be an iter like that tho

Reply:
why not?

Reply:
had a brain fart, ye it can